main.go

```diff
@@ -63,7 +63,17 @@ func main() {
 	logg.ShowDebug, _ = strconv.ParseBool(os.Getenv("CASTELLUM_DEBUG"))
 
 	//initialize DB connection
-	dbi, err := db.Init(mustGetenv("CASTELLUM_DB_URI"))
+	dbUserInfo := envOrDefault("CASTELLUM_DB_USERNAME", "postgres")
+	if dbPass := os.Getenv("CASTELLUM_DB_PASSWORD"); dbPass != "" {
+		dbUserInfo += ":" + dbPass
+	}
+	dbHost := envOrDefault("CASTELLUM_DB_HOSTNAME", "localhost")
+	dbPort := envOrDefault("CASTELLUM_DB_PORT", "5432")
+	dbName := envOrDefault("CASTELLUM_DB_NAME", "castellum")
+	dbConnOpts := envOrDefault("CASTELLUM_DB_CONNECTION_OPTIONS", "sslmode=disable")
+
+	dbURL := fmt.Sprintf("postgres://%s@%s:%s/%s?%s", dbUserInfo, dbHost, dbPort, dbName, dbConnOpts)
+	dbi, err := db.Init(dbURL)
```

Comment:
Since we have all the bits and pieces of the URL already, it's weird to merge it all together into a string, then parse it again. `db.Init` should take a `*url.URL` argument which `main` constructs as a literal: `dbURL := &url.URL { Host: dbHost + ":" + dbPort, ... }`.